hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -242,6 +244,8 @@ public HazelcastClientInstanceImpl(ClientConfig config,
         lockReferenceIdGenerator = new ClientLockReferenceIdGenerator();
         nearCacheManager = clientExtension.createNearCacheManager();
         clientExceptionFactory = initClientExceptionFactory();
+
+        statistics = new Statistics(this);
```

Comment:
Be careful with passing this in a constructor to another object. Easy to run into a partial constructed object (one of the Effective Java anti patterns)